kubernetes/pod/build.go

```diff
@@ -209,3 +209,69 @@ func (b *Builder) WithVolumeBuilders(volumeBuilderList []*volume.Builder) *Build
 	}
 	return b
 }
+
+// WithTolerations merges the existing tolerations
+// with the provided arguments
+func (b *Builder) WithTolerations(tolerations ...corev1.Toleration) *Builder {
+	if tolerations == nil {
+		b.errs = append(
+			b.errs,
+			errors.New(
+				"failed to build pod object: nil tolerations",
+			),
+		)
+		return b
+	}
+	if len(tolerations) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New(
+				"failed to build pod object: missing tolerations",
+			),
+		)
+		return b
+	}
+
+	if len(b.pod.object.Spec.Tolerations) == 0 {
+		return b.WithTolerationsNew(tolerations...)
```

Comment:
Can we use `b.pod.object.Spec.Tolerations = tolerations` here directly?

Reply:
We can - however we use the New() fn in case no tolerations are defined by that point. This one is more of an append func. In case of engines, it is always a fresh toleration anyways - but this kind of separation is good for a future scenario where there are being used in builders for objects already having some default tolerations.